src/modm/driver/temperature/max31865.hpp

```diff
@@ -0,0 +1,236 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Henrik Hose
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MAX31865_HPP
+#define MODM_MAX31865_HPP
+
+#include <array>
+#include <modm/architecture/interface/register.hpp>
+#include <modm/architecture/interface/spi_device.hpp>
+#include <modm/processing/resumable.hpp>
+#include <modm/processing/timer.hpp>
+
+namespace modm
+{
+
+struct Pt
+{
+	float R0;
+	float Rref;
+	float alpha;
+	double a;
+	double b;
+	double c;
+};
+
+/// Standard values for IEC 751 (PT100) with 430 Ohm reference
+constexpr Pt pt100{
+	.R0 = 100.f,
+	.Rref = 430.f,
+	.alpha = 3.85055e-3f,
+	.a = double{3.90830e-3},
+	.b = double{-5.77500e-7},
+	.c = double{-4.18301e-12},
+};
```

Comment:
I don't like that this pollutes the `modm::` namespace with `Pt` and `pt100`, which are not obvious without the context. They should be moved inside the non-template `max31865` struct.